### PR TITLE
Feature/#37

### PR DESCRIPTION
**Pull request by jamesfoote**
Finished Feature/#37.

Notable changes are:
-Addition of creation and updating to kubeconfig automatically
-Addition of waiting for a cluster to finish preparing before returning on create_instance
-Addition of a basic example usage of the GKE_Client
-Addition of a environment variable when credentials
-Minor bug fixes in GKE and exceptions

### CODE REVIEW

File: src/ucapi/gke_client.py [hunk 1]
```diff
@@ -103,18 +106,35 @@ def __init__(self, credentials, region, project_id):
...
+        :param created_clusters: all the clusters created in the project
+        :param cluster_name: name of the project being checked
+        :return: Boolean
+        """
+        if created_clusters is None:
+            return False
+        else:
+            for cluster in created_clusters["clusters"]:
+                if cluster["name"] == cluster_name and cluster["status"] == "RUNNING":
+                    print(cluster["status"])
```
Comment (BlairTwiss) [inline src/ucapi/gke_client.py]: Is this print statement still needed?

File: src/ucapi/gke_client.py [hunk 2]
```diff
@@ -123,9 +143,17 @@ def create_cluster(self, cluster_name):
                 }
             }
 
-            # create_cluster
-            return self._gke_container_client.projects().zones().clusters().create(projectId=self._project_id,
-                                                                                   zone=self._zone, body=body).execute()
+            # Creates the Cluster
+            print("Creating cluster.")
```
Comment (BlairTwiss) [inline src/ucapi/gke_client.py]: Is this print statement still needed?

File: src/ucapi/gke_client.py [hunk 3]
```diff
@@ -123,9 +143,17 @@ def create_cluster(self, cluster_name):
...
-            # create_cluster
-            return self._gke_container_client.projects().zones().clusters().create(projectId=self._project_id,
-                                                                                   zone=self._zone, body=body).execute()
+            # Creates the Cluster
+            print("Creating cluster.")
+            response = self._gke_container_client.projects().zones().clusters().create(projectId=self._project_id,
+                                                                                       zone=self._zone,
+                                                                                       body=body).execute()
+            # Doesn't return from function until the cluster is fully created
+            while not self._check_cluster_status(self._get_clusters(), cluster_name):
```
Comment (BlairTwiss) [inline src/ucapi/gke_client.py]: Potentially add a time out so people don't get stuck in an endless while loop?

File: src/ucapi/gke_client.py (hunk 2 above)
Comment (jamesfoote) [inline src/ucapi/gke_client.py]: I think leave this one for now, but I deleted the other one as it's not really necessary.

**Pull request merged**